lhotse/array.py

```diff
@@ -121,9 +121,13 @@ def shape(self) -> List[int]:
     def ndim(self) -> int:
         return len(self.shape)
 
+    @property
+    def num_frames(self) -> int:
+        return self.shape[self.temporal_dim]
+
     @property
     def duration(self) -> Seconds:
-        return self.shape[self.temporal_dim] * self.frame_shift
+        return self.num_frames * self.frame_shift
```

Comment:
Is `frame_shift` supposed to be different from the one contained in `cut.frame_shift`?

Reply:
Originally, `cut` was designed to have only features and recording, so there was only one `frame_shift`. But now, because the custom fields are generic, you can put as many arrays with different frame shifts as you want. 

Lhotse doesn't expect any relationship between `cut.frame_shift` and a custom field's `frame_shift`. It is completely up to the user.

Reply:
Just to give a specific example:

`cut.features` -> `frame_shift=0.01`
`cut.phone_alignment` -> `frame_shift=0.04`
`cut.online_ivectors` -> `frame_shift=0.1`